shmem.c

```diff
@@ -346,8 +406,9 @@ bool realloc_shm(SharedMemory *sharedMemory, size_t size) {
 	}
 
 	// Close shared memory object file descriptor as it is no longer
-	// needed after having called mmap()
-	close(fd);
+	// needed after having called ftruncate()
+	if(resize)
+		close(fd);
```

Comment:
The file descriptor could be closed at the end of the first if block where we resize the memory, instead of waiting until here to close it. It's only used in a logging statement outside of the if block (and doesn't need to be there).